src/NHSD.BuyingCatalogue.Ordering.Api/Services/CreatePurchasingDocument/CreatePurchasingDocumentService.cs

```diff
@@ -45,7 +49,43 @@ public async Task CreatePatientNumbersCsvAsync(Stream stream, Order order)
                 M1Planned = orderItem.DeliveryDate,
             }).ToList();
 
-            await _patientNumbersCsvStreamWriter.WriteRecordsAsync(stream, patientNumbersPriceTypes);
+            await _patientNumbersCsvWriter.WriteRecordsAsync(stream, patientNumbersPriceTypes);
+        }
+
+        public async Task CreatePriceTypeCsvAsync(Stream stream, Order order)
+        {
+            if (stream is null)
+                throw new ArgumentNullException(nameof(stream));
+
+            if (order is null)
+                throw new ArgumentNullException(nameof(order));
+
+            var serviceRecipientDictionary = order.ServiceRecipients.ToDictionary(x => x.OdsCode);
+
+            var priceType = order.OrderItems.Select(orderItem => new PriceType
+            {
+                CallOffAgreementId = order.OrderId,
+                CallOffOrderingPartyId = order.OrganisationOdsCode,
+                CallOffOrderingPartyName = order.OrganisationName,
+                CallOffCommencementDate = order.CommencementDate,
+                ServiceRecipientId = orderItem.OdsCode,
+                ServiceRecipientName = serviceRecipientDictionary[orderItem.OdsCode]?.Name,
+                ServiceRecipientItemId = $"{orderItem.OrderItemId}-{orderItem.OdsCode}-{orderItem.OrderItemId}",
```

Comment:
is this right? it looks weird. OrderItemId is in there twice

Reply:
Well spotted, just triple checked and it should be 
ServiceRecipientItemId = $"{order.OrderId}-{orderItem.OdsCode}-{orderItem.OrderItemId}"

Updated the one above too